backup

```diff
@@ -54,10 +66,21 @@ restic --repo="${RESTIC_REPOSITORY}" backup "${backup_args[@]}" "${tag_options[@
 rc=$?
 set -e
 
+success="true"
 if [ $rc -ne 0 ]; then
-  if [ $rc -eq 3 ] && [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
-  	run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+  if [ $rc -eq 3 ]; then
+    if [ "$(check_bool "${SUCCESS_ON_INCOMPLETE_BACKUP:-}")" = "true" ]; then
+      run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+    else
+      success="false"
+      if [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
+        run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+      else
+        run_commands "${POST_COMMANDS_FAILURE:-}"
+      fi
+    fi
   else
+    success="false"
   	run_commands "${POST_COMMANDS_FAILURE:-}"
   fi
 fi
```

Comment:
Could this be simplified to something like;

```bash
success="true"
if [ $rc -ne 0 ]; then
  if [ $rc -eq 3 ]; then
    run_commands "${POST_COMMANDS_INCOMPLETE:-}"
    success="$(check_bool "${SUCCESS_ON_INCOMPLETE_BACKUP:-}")"
  else
    success="false"
  fi
fi

if [ "$success" == "true" ]; then
  echo Backup successful
else
  echo Backup failed
fi

if [ -n "${RESTIC_FORGET_ARGS}" ]; then
  IFS=" " read -r -a RESTIC_FORGET_ARGS <<< "$RESTIC_FORGET_ARGS"
  echo Forget about old snapshots based on RESTIC_FORGET_ARGS = "${RESTIC_FORGET_ARGS[@]}"
  restic forget "${tag_options[@]}" "${RESTIC_FORGET_ARGS[@]}"
fi

end=$(date +%s)
echo Finished backup at $(date +"%Y-%m-%d %H:%M:%S") after $((end-start)) seconds

if [ "$success" == "true" ]; then
  run_commands "${POST_COMMANDS_SUCCESS:-}"
else
  run_commands "${POST_COMMANDS_FAILURE:-}"
fi
```

Reply:
Thanks, that was a good inspiration for a refactoring which I now did in 2456f18ad4714cdaae68e14a16cd27c2e5bed193.

Reply:
Nice - is it passing your internal testing now? I am unable to test currently sorry.

Reply:
The shellspec tests are still passing. Now need to test IRL.

Reply:
Sorry to keep bugging you @djmaze - how has your IRL testing been going?

Reply:
@sumnerboy12 Sorry for the late reply. I have connection problems on my test cluster which I need to fix first.